app/controllers/sessions_controller.rb

```diff
@@ -2,12 +2,14 @@ class SessionsController < ApplicationController
   def callback
     auth = request.env['omniauth.auth']
 
-    user = User.find_by(google_uid: auth[:uid])
-    unless user
-      user = User.create(google_uid: auth[:uid], name: auth[:info][:name], email: auth[:info][:email], image_url: auth[:info][:image])
+    if auth.info['email'].split('@')[1] == ENV['RESTRICT_DOMAIN']
+      user = User.form_omniauth(auth) unless user = User.find_by(google_uid: auth[:uid])
```

Comment:
__[rubocop]__
* Assignment in condition - you probably meant to use `==`. [:ref](https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
* Line is too long. [88/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)